dialogue-core/src/test/java/com/palantir/dialogue/core/UrlBuilderTest.java

```diff
@@ -160,11 +160,11 @@ public void urlEncoder_encodePathSegment_onlyEncodesNonReservedChars() {
 
     @Test
     public void urlEncoder_encodeQuery_onlyEncodesNonReservedChars() {
-        String nonReserved = "aAzZ09!$'()*+,;?/";
+        String nonReserved = "aAzZ09!$'()*,;/?";
```

Comment:
please add an explicit test that shows how we differ from the RFC

Reply:
Not sure what you're suggesting. An explicit test would be redundant. This test clearly demonstrates which characters are percent-encoded and which are not.

Reply:
```
@Test
public void urlEncoder_encodeQuery_encodesPlusSign() {
    // the encode behavior differs from the RFC in that it protects against servers that decode + as space
    ...
}
```

Reply:
it's just a service to future readers/maintainers of this code. deviation from the spec is very surprising and I want to make sure (in two places) that we documented that and why we do it.

Reply:
Explicit test for behavior where we've decided to take a more specific approach sounds reasonable, though I don't think it differs from the RFC, it chooses a specific path where the rfc gives us the option not to encode `+`.

It would be helpful to call out specifically in the test that support among other libraries is mixed for handling non-encoded `+` in case someone wants to change the code later, they'll be more likely to reconsider :-)